src/app/store/shopping-list/shopping-list.effects.ts

```diff
@@ -149,6 +149,32 @@ export class shoppingListEffects {
 		);
 	});
 
+	updateIngredientsFromRecipe = createEffect(() => {
+		return this.actions$.pipe(
+			ofType(shoppingListActions.updateIngredient),
+			concatLatestFrom(() =>
+				this.store.select(ShoppingListSelectors.selectAll)
+			),
+			map(([action, ingredientInShoppingList]: [any, IngredientsInStore[]]) => {
+				let ingredientsToStore = ingredientInShoppingList;
+				const { ingredient, previewIngredient } = action;
+
+				ingredientsToStore = ingredientsToStore.filter(
+					ingredient => ingredient !== previewIngredient
+				);
+
+				ingredientsToStore = this.ingredientService.addIngredientAndGroupe(
```

Comment:
is this synchronous function?

Reply:
I use local storage to keep my ingredients